api/internal/features/notification/controller/get_smtp.go

```diff
@@ -46,6 +47,14 @@ func (c *NotificationController) GetSmtp(f fuego.ContextNoBody) (*shared_types.R
 
 	SMTPConfigs, err := c.service.GetSmtp(user.ID.String(), orgID)
 	if err != nil {
+		if errors.Is(err, notification.ErrSMTPConfigNotFound) {
```

Comment:
can you generalize both not found conditions, there is one more if statement that checks for if smtpconfigs not present,  code duplication